src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -9,18 +9,148 @@
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private boolean isEmpty(int arrayIndex) {
+        return Objects.isNull(nodeArray[arrayIndex]);
+    }
+
+    private void reSize() {
+        int newArraySize = nodeArray.length * 2;
+        Node<K, V>[] newNodeArray = (Node<K, V>[]) new Node[newArraySize];
+        thresholdCalculation(newArraySize);
+        size = 0;
+        Node<K, V>[] oldNodeArray = nodeArray;
+        nodeArray = newNodeArray;
+        for (Node<K, V> oldNode : oldNodeArray) {
+            if (Objects.nonNull(oldNode)) {
+                newArrayIndex = findIndex(oldNode.key);
+                addNewNode(oldNode.key, oldNode.value, newArrayIndex);
+                Node<K, V> nextNode = oldNode.next;
+                while (nextNode != null) {
+                    int newNodeIndex = findIndex(nextNode.key);
+                    addNewNode(nextNode.key, nextNode.value, newNodeIndex);
+                    nextNode = nextNode.next;
+                }
+            }
+        }
+    }
+
+    private int findIndex(K key) {
+        return Objects.isNull(key) ? 0 : Math.abs(hash(key) % nodeArray.length);
+    }
+
+    private void addNewNode(K key, V value, int arrayIndex) {
+        Node<K, V> nodeByIndex = nodeArray[arrayIndex];
+        if (isEmpty(arrayIndex)) {
+            if (Objects.isNull(key)) {
+                nodeArray[arrayIndex] = new Node<>(null, value, 0, null);
+                size++;
+                return;
+            }
+            nodeArray[arrayIndex] = new Node<>(key, value, hash(key), null);
+            size++;
+            return;
+        }
+        if (Objects.equals(nodeByIndex.key, key)) {
+            nodeByIndex.value = value;
+            return;
+        }
+        Node<K, V> nextNode = nodeByIndex.next;
+        do {
+            if (Objects.isNull(nextNode)) {
+                if (Objects.isNull(key)) {
+                    nodeByIndex.next = new Node<>(null, value, 0, null);
+                    size++;
+                    return;
+                }
+                nodeByIndex.next = new Node<>(key, value, hash(key), null);
+                size++;
+                return;
+            }
+            if (Objects.equals(nextNode.key, key)) {
+                nextNode.value = value;
+                return;
+            }
+            if (Objects.isNull(nextNode.next)) {
+                if (Objects.isNull(key)) {
+                    nextNode.next = new Node<>(null, value, 0, null);
+                    size++;
+                    return;
+                }
+                nextNode.next = new Node<>(key, value, hash(key), null);
+                size++;
+                return;
+            }
+            nextNode = nextNode.next;
+        } while (true);
+    }
+
+    private int thresholdCalculation(int ArraySize) {
+        return threshold = (int) (ArraySize * LOAD_FACTOR);
```

Comment:
never do like that, just return calculated value and assign result in variable outside the method

Reply:
@al-pavlenko +